src/edit.js

```diff
@@ -90,9 +115,16 @@ const userDropdown = () => {
 		<>
 			<InspectorControls>
                 <PanelBody title="Users">
-                    <DropdownMenu
+                    <SelectControl
                         label="Users"
-                        controls={ users.map(user => ( {title: user.name, onClick: userDropdown } ))}
+                        options={ mappedUsers }
+                        value={ selectedUser?.id || 0 }
+                        onChange={ ( userId ) => {
+                            const userFromUsers = mappedUsers.find( ( user ) => {
+                                return user.id === parseInt( userId );
+                            } );
```

Comment:
The array method `find` returns the first value found to match the condition provided (here, when `user.id` matches `userId`).

You could also `filter`, but then you would need to parse that array as well.